### PR TITLE
[Transit] Pull and Push Root Snapshot

### PR DESCRIPTION
**Pull request by danuio**
Resolves https://github.com/dapperlabs/flow-go/issues/5288.

### CODE REVIEW

File: cmd/bootstrap/transit/main.go [hunk 1]
```diff
@@ -283,11 +280,10 @@ func runDownloadSnapshot(ctx context.Context, bootDir, nodeIDHex, accessAddress
...
 	}
 
-	// unmarshal bytes to snapshot object
-	var snapshot inmem.Snapshot
-	err = json.Unmarshal(bytes, &snapshot)
+	// unmarshal bytes to snapshot
+	snapshot, err := convert.BytesToInmemSnapshot(bytes)
 	if err != nil {
-		log.Fatal("could not unmarshal snapshot data retreived from access node")
+		log.Fatal("could not convert array of bytes to snapshot")
```
Comment (vishalchangrani) [inline cmd/bootstrap/transit/main.go:286]:
```suggestion
		log.Fatalf("could not convert array of bytes to snapshot: %v", err)
```

File: engine/common/rpc/convert/convert.go [hunk 2]
```diff
@@ -370,10 +370,21 @@ func SnapshotToBytes(snapshot protocol.Snapshot) ([]byte, error) {
...
 
 	return data, nil
 }
+
+// BytesToInmemSnapshot converts an array of bytes to `inmem.Snapshot`
+func BytesToInmemSnapshot(bytes []byte) (*inmem.Snapshot, error) {
+	var snapshot inmem.Snapshot
+	err := json.Unmarshal(bytes, &snapshot)
+	if err != nil {
+		return nil, fmt.Errorf("could not unmarshal snapshot data retreived from access node")
```
Comment (vishalchangrani) [inline engine/common/rpc/convert/convert.go:386]:
```suggestion
		return nil, fmt.Errorf("could not unmarshal snapshot data retreived from access node: %v", err)
```

**Review by vishalchangrani — APPROVED** [review]

File: cmd/bootstrap/transit/main.go [hunk 3]
```diff
@@ -45,6 +44,7 @@ var (
 	// default files to upload for all role type
 	filesToUpload = []string{
 		bootstrap.PathNodeInfoPub,
+		bootstrap.PathRootProtocolStateSnapshot, // push root protocol state snapshot
```
Comment (jordanschalm) [inline cmd/bootstrap/transit/main.go:47]: If previously this logic uploaded the root block, seal, etc. files, and we are now writing the root-snapshot file in the same directory as we previously wrote those files as of https://github.com/onflow/flow-go/pull/424, then I don't think this is correct. The file tree would look like:
```
- $BOOTDIR/PathNodeInfoPub
|
+- root-snapshot.json
```

where previously it looked like:
```
- $BOOTDIR/PathNodeInfoPub
|
+- /root-block.json
+- /root-seal.json
+- /root-result.json
```

meaning, we still only need to upload the `PathNodeInfoPub` directory.

**Review by jordanschalm — APPROVED** [review]
Just one note, otherwise looks good!

**Pull request merged**